inc/gpio.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+// GPIO HAL Interface
+
+#include <stdbool.h>
+#include <stdint.h>
+
+// IO Pin "address" to be used to change that pin's settings
+// Use bytes instead of words to save on memory also convenient for MSP430
+// compatibility
+typedef struct IOAddr {
+  const uint8_t IOPort;  // Could be an enum (A-F) for stm32f0xx
+  const uint8_t IOPin;
+} IOAddr;
+
+// For setting the direction of the pin
+typedef enum { PIN_IN = 0; PIN_OUT; }
+PIN_DIR;
+
+// For setting the output value of the pin
+typedef enum { PIN_LOW = 0; PIN_HIGH; }
+PIN_VAL;
```

Comment:
I've generally seen this as "state" vs. "val". Same thing with naming and prefixing.